### PR TITLE
feat: separate musl from targets, native downloader and ci update 

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

* [x] Implement file downloader with retries and timeouts
* [x] Unify CI workflows removing boilerplate using one matrix job instead of 3
* [x] Add `--musl` option for `build` and `clone` allowing building and testing of LLVM Musl targets regardless of the LLVM builder build target
* [x] Make `VERBOSE` and `DRY_RUN` configurable from environment variables

## Why ❔

* To fix sporadic Musl download failures across all workflows
* To ease CI support in the future
* To simplify installation instructions for LLVM builder and allow Musl and Gnu builds with the same standard LLVM builder binaries

## Checklist

- [x] PR title corresponds to the body of PR.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: src/zkevm_llvm/arguments.rs [hunk 1]
```diff
@@ -15,12 +15,18 @@ pub enum Arguments {
...
+        use_musl: bool,
     },
     /// Build the LLVM framework.
     Build {
         /// Whether to build the 'Debug' version.
         #[structopt(long = "debug")]
         debug: bool,
+        /// Build LLVM with musl.
+        #[structopt(long = "musl")]
+        use_musl: bool,
```
Comment (hedgar2017) [inline src/zkevm_llvm/arguments.rs:29]: Let's make an enum `gnu | musl`?

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: src/zkevm_llvm/arguments.rs (hunk 1 above)
Comment (antonbaliasnikov) [reply src/zkevm_llvm/arguments.rs]: Done, please, take a look

**Comment by antonbaliasnikov** [comment]
@hedgar2017 , we need to find a proper way to merge this, because it will require changes in CI in all repos that use builder. 

I suggest to do it as follows:
1. Make sure we release the current `main` to the cargo crate
2. Update other repositories to use LLVM builder from cargo mirror, not from GitHub (as it is now to always take the most recent version)
3. Merge this PR and release a new version to cargo
4. Update other repositories CI with the new version of builder from cargo and update docs at the same time - this will let us to properly test our changes

**Comment by hedgar2017** [comment]
> @hedgar2017 , we need to find a proper way to merge this, because it will require changes in CI in all repos that use builder.
> 
> I suggest to do it as follows:
> 
> 1. Make sure we release the current `main` to the cargo crate
> 2. Update other repositories to use LLVM builder from cargo mirror, not from GitHub (as it is now to always take the most recent version)
> 3. Merge this PR and release a new version to cargo
> 4. Update other repositories CI with the new version of builder from cargo and update docs at the same time - this will let us to properly test our changes

Sure! Let's do it now.
Will publish in 5 min.

**Pull request merged**